### PR TITLE
feat: Add SkipLink component

### PR DESCRIPTION
**Pull request by ActuallyACat**
# Objective
Add a SkipLink component to the a11y package. 

Related to this PR (uses the same package): https://github.com/cultureamp/kaizen-design-system/pull/2200
Closes: https://github.com/cultureamp/team-foundations/issues/15

# Motivation and Context
* Keyboard users with limited mobility will waste time and energy navigating past elements that repeat on all pages.
* Screen reader users will get frustrated listening to the same banner and navigation content on every page they visit.

# Checklist

- [ ] If this contains visual changes, has it been reviewed by a designer?
- [x] If this introduces either a new component or a breaking change to an existing component, has it been reviewed by an advocate? (Ask in #prod_design_systems)
- [x] I have considered likely risks of these changes and got someone else to QA as appropriate
- [x] I have or will communicate these changes to the front end practice

https://user-images.githubusercontent.com/1621182/140014738-1fdfdd18-3e14-498e-83b9-b670fda32c37.mov

### CODE REVIEW

File: packages/a11y/package.json [hunk 1]
```diff
@@ -0,0 +1,40 @@
+{
+  "name": "@kaizen/a11y",
```
Comment (lloydstubber) [inline packages/a11y/package.json:2]: yeah it is

Comment (ActuallyACat) [reply packages/a11y/package.json:2]: 👉🏻 😎 👉🏻

**Review by lloydstubber — DISMISSED** [review]
Love this! Nice work

File: packages/a11y/docs/SkipLink.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+  Primary,
+  ArgsTable,
+  Stories,
+  PRIMARY_STORY,
+} from "@storybook/addon-docs"
+import { CATEGORIES } from "../../../storybook/constants"
+import { SkipLink } from "../index"
+
+export default {
+  title: `${CATEGORIES.components}/SkipLink`,
```
Comment (dougmacknz) [inline packages/a11y/docs/SkipLink.stories.tsx:16]: (nitpick sorry)

```suggestion
  title: `${CATEGORIES.components}/Skip Link`,
```

File: packages/a11y/docs/SkipLink.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+  component: SkipLink,
+  parameters: {
+    docs: {
+      description: {
+        component: 'import { SkipLink } from "@kaizen/component-library"',
```
Comment (dougmacknz) [inline packages/a11y/docs/SkipLink.stories.tsx:21]:
```suggestion
        component: 'import { SkipLink } from "@kaizen/a11y"',
```

File: packages/a11y/package.json [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+  "scripts": {
+    "prepublish": "tsc --project tsconfig.dist.json",
+    "build": "yarn clean && yarn prepublish",
+    "build:watch": "yarn clean && yarn prepublish --watch",
+    "clean": "rimraf '**/*.d.ts' '**/*.js' '**/*.map'"
+  },
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/cultureamp/kaizen-design-system.git",
+    "directory": "packages/brand"
```
Comment (dougmacknz) [inline packages/a11y/package.json:14]:
```suggestion
    "directory": "packages/a11y"
```

File: packages/a11y/package.json [hunk 5]
```diff
@@ -0,0 +1,40 @@
+{
+  "name": "@kaizen/a11y",
+  "version": "0.0.2",
```
Comment (dougmacknz) [inline packages/a11y/package.json:3]: interesting version to start with 😅

File: packages/a11y/src/Skiplink.spec.tsx [hunk 6]
```diff
@@ -0,0 +1,15 @@
+import { cleanup, render } from "@testing-library/react"
+import * as React from "react"
+import { SkipLink } from "./SkipLink"
+
+afterEach(cleanup)
```
Comment (dougmacknz) [inline packages/a11y/src/Skiplink.spec.tsx:5]: FYI you don't need to do this anymore: https://kentcdodds.com/blog/common-mistakes-with-react-testing-library#using-cleanup

File: packages/a11y/src/Skiplink.spec.tsx [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+
+describe("<SkipLink />", () => {
+  it("renders a SkipLink link as an anchor tag", () => {
```
Comment (dougmacknz) [inline packages/a11y/src/Skiplink.spec.tsx:8]: seems like a bit of a strange test, in no situations would this ever _not_ be an anchor?

Comment (dougmacknz) [reply packages/a11y/src/Skiplink.spec.tsx:8]: Guessing the answer is no, but Is there a way that we could test the actually functionality?

File: packages/a11y/package.json (hunk 5 above)
Comment (ActuallyACat) [reply packages/a11y/package.json:3]: 😬 I was testing this in Unified Navigation and each time I published a change I needed a new version.

File: packages/a11y/src/Skiplink.spec.tsx (hunk 6 above)
Comment (ActuallyACat) [reply packages/a11y/src/Skiplink.spec.tsx:5]: TIL - thanks!

File: packages/a11y/package.json (hunk 5 above)
Comment (ActuallyACat) [reply packages/a11y/package.json:3]: It doesn't _super_ matter what version this package starts with, as Lerna will publish it as `1.1.0`

Comment (dougmacknz) [reply packages/a11y/package.json:3]: not 0.1.0? (not that I have any issues with that)

File: packages/a11y/src/Skiplink.spec.tsx (hunk 7 above)
Comment (ActuallyACat) [reply packages/a11y/src/Skiplink.spec.tsx:8]: Hmm... I can test that it receives focus, but I feel ick about testing if it's visuallyHidden with styles.

File: package.json [hunk 8]
```diff
@@ -64,6 +64,7 @@
     "@testing-library/jest-dom": "^5.11.4",
     "@testing-library/react": "^11.2.3",
     "@testing-library/react-hooks": "^4.0.0",
+    "@testing-library/user-event": "^13.5.0",
```
Comment (dougmacknz) [inline package.json:67]: It finally happened 🎉

File: packages/a11y/src/Skiplink.spec.tsx [hunk 9]
```diff
@@ -0,0 +1,16 @@
...
+import { SkipLink } from "./SkipLink"
+import "@testing-library/jest-dom"
+
+describe("<SkipLink />", () => {
+  it("SkipLink is focusable", () => {
+    const { getByText } = render(
+      <SkipLink label="Skip to main content" skipTo="mainContent" />
+    )
+    expect(document.body).toHaveFocus()
+    userEvent.tab()
```
Comment (dougmacknz) [inline packages/a11y/src/Skiplink.spec.tsx:13]: Awesome

**Review by dougmacknz — APPROVED** [review]

**Pull request merged**